server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/task/common/DataTransferTracer.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.oceanbase.odc.service.datatransfer.task.common;
+
+import java.util.List;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import com.oceanbase.odc.core.shared.constant.TaskStatus;
+import com.oceanbase.odc.service.datatransfer.model.DataTransferScope;
+import com.oceanbase.odc.service.datatransfer.task.TransferTask;
+import com.oceanbase.odc.service.flow.task.model.DataTransferTaskResult;
+
+public class DataTransferTracer {
+
+    private final DataTransferTaskRunner job;
+    private final Future<Void> future;
+    private volatile TaskStatus status = TaskStatus.RUNNING;
+
+    public DataTransferTracer(DataTransferTaskRunner job, Future<Void> future) {
+        this.job = job;
+        this.future = future;
+    }
+
+    public DataTransferTaskResult get() {
+        DataTransferTaskResult result = new DataTransferTaskResult();
+
+        job.getTaskQueue().forEach(task -> {
+            if (task.scope() == DataTransferScope.DATA) {
```

Comment:
why should we have to know which task transfers data, which task transfer schema?